server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -785,6 +786,13 @@ public static String toString(ToXContent toXContent) {
         return toString(toXContent, false, false);
     }
 
+    /**
+     * Return a {@link String} that is the json representation of the provided {@link ChunkedToXContent}.
+     */
+    public static String toString(ChunkedToXContent chunkedToXContent) {
```

Comment:
Can we avoid this and the following method? Seems like bugs if we invoke them, given that they produce Strings that could be too large to handle?

Reply:
I agree, this sucks. But it's not a new issue and IMO this change at least makes the problem areas clear. The `toString` output is unchanged from what it was. Maybe a good move would be to just deprecate these outright and then fix them in a follow up by changing the `toString` output. This isn't entirely trivial since we'd have to come up with a new format for each implementation on a case by case basis so I'd rather not have those discussions here I think :)

Reply:
done added deprecation :)